dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/orm/EntityParser.java

```diff
@@ -0,0 +1,747 @@
+import java.io.Serializable;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.RecordComponent;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Trivial;
+
+import io.openliberty.data.internal.persistence.Util;
+import io.openliberty.data.internal.persistence.orm.Models.AccessType;
+import io.openliberty.data.internal.persistence.orm.Models.Attribute;
+import io.openliberty.data.internal.persistence.orm.Models.AttributeKind;
+import io.openliberty.data.internal.persistence.orm.Models.Converter;
+import io.openliberty.data.internal.persistence.orm.Models.EmbeddableRecord;
+import io.openliberty.data.internal.persistence.orm.Models.EntityRecord;
+import io.openliberty.data.internal.persistence.orm.Models.MappedSuperclass;
+import jakarta.data.exceptions.MappingException;
+import jakarta.persistence.AttributeConverter;
+import jakarta.persistence.Convert;
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Embedded;
+import jakarta.persistence.EmbeddedId;
+import jakarta.persistence.Entity;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+
+/**
+ * Parses entity(ies) and creates an Object-Relational Model in memory
+ * that includes managed objects that do not have the Jakarta Persistence
+ * annotations required to be automatically modeled by the underlying Jakarta
+ * Persistence provider.
+ *
+ * This is an add-on feature of the Liberty Jakarta Data provider to support
+ * record entities which cannot be annotated, and for users who write Repository
+ * classes for java objects that comes from third party libraries which cannot
+ * be annotated.
+ */
+public class EntityParser {
+    private static final TraceComponent tc = Tr.register(EntityParser.class);
+
+    // ORM sets
+    private final Map<Class<?>, MappedSuperclass> mappedSuperclasses;
+    private final Map<Class<?>, EntityRecord> entities;
+    private final Map<Class<?>, EmbeddableRecord> embeddables;
+    private final Map<Class<?>, Converter> converters;
+
+    // Relationship tracking
+    private final Map<Class<?>, Class<?>> entityToRecord;
+    private final ConcurrentMap<Class<?>, Set<Class<?>>> entitiesSuperclasses;
```

Comment:
Is there a reason this needed to be a ConcurrentMap? I don't see anywhere it is modified/used by multiple threads at once,
```suggestion
    private final Map<Class<?>, Set<Class<?>>> entitiesSuperclasses;
```